lib/app/features/users/data/repositories/users_repository.dart

```diff
@@ -68,6 +71,35 @@ class UsersRepository implements IUsersRepository {
       return left(MapExceptionToFailure.map(error));
     }
   }
+
+  @override
+  Future<Either<Failure, ValidField>> report(
+      String clientId, String reason) async {
+    try {
+      const endPoint = '/report-profile';
+      final parameters = {'reason': reason, 'cliente_id': clientId};
+      final response = await _apiProvider!
+          .post(path: endPoint, parameters: parameters)
+          .parseValidField();
+      return right(response);
+    } catch (error, stack) {
+      logError(error, stack);
+      return left(MapExceptionToFailure.map(error));
+    }
+  }
+
+  @override
+  Future<Either<Failure, ValidField>> block(String clientId) async {
+    try {
+      const endPoint = '/profile-block';
+      final response = await _apiProvider!.post(
+          path: endPoint,
+          parameters: {'cliente_id': clientId}).parseValidField();
+      return right(response);
+    } catch (error) {
+      return left(MapExceptionToFailure.map(error));
+    }
```

Comment:
seria interessante fazer o log do erro:
```suggestion
    } catch (error, stack) {
      logError(error, stack);
      return left(MapExceptionToFailure.map(error));
    }
```